presto-iceberg/src/main/java/com/facebook/presto/iceberg/delete/IcebergDeletePageSink.java

```diff
@@ -0,0 +1,220 @@
+        this.partitionSpec = requireNonNull(partitionSpec, "partitionSpec is null");
+        this.partitionData = partitionData;
+        this.locationProvider = requireNonNull(locationProvider, "locationProvider is null");
+        this.fileWriterFactory = requireNonNull(fileWriterFactory, "fileWriterFactory is null");
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.hdfsContext = requireNonNull(hdfsContext, "hdfsContext is null");
+        this.jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(locationProvider.newDataLocation("delete-file"))));
+        this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
+        this.session = requireNonNull(session, "session is null");
+        this.dataFile = requireNonNull(dataFile, "dataFile is null");
+        this.fileFormat = requireNonNull(fileFormat, "fileFormat is null");
+
+        String fileName = fileFormat.addExtension(String.format("delete_file_%s", randomUUID().toString()));
+        this.outputPath = partitionData.map(partition -> new Path(locationProvider.newDataLocation(partitionSpec, partition, fileName)))
+                .orElse(new Path(locationProvider.newDataLocation(fileName)));
+        this.positionDeleteWriter = new IcebergPositionDeleteWriter();
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return writtenBytes;
+    }
+
+    @Override
+    public long getSystemMemoryUsage()
+    {
+        return systemMemoryUsage;
+    }
+
+    @Override
+    public long getValidationCpuNanos()
+    {
+        return validationCpuNanos;
+    }
+
+    @Override
+    public CompletableFuture<?> appendPage(Page page)
+    {
+        positionDeleteWriter.appendPage(page);
+        return NOT_BLOCKED;
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        Collection<Slice> commitTasks = new ArrayList<>();
+
+        IcebergFileWriter writer = positionDeleteWriter.getWriter();
+
+        writer.commit();
+
+        CommitTaskData task = new CommitTaskData(
+                outputPath.toString(),
+                writer.getFileSizeInBytes(),
+                new MetricsWrapper(writer.getMetrics()),
+                partitionData.map(PartitionData::toJson),
+                fileFormat,
+                dataFile);
+
+        commitTasks.add(wrappedBuffer(jsonCodec.toJsonBytes(task)));
+
+        writtenBytes = writer.getWrittenBytes();
+        validationCpuNanos = writer.getValidationCpuNanos();
+
+        return completedFuture(commitTasks);
+    }
+
+    @Override
+    public void abort()
+    {
+        RuntimeException error = null;
+        try {
+            positionDeleteWriter.getWriter().rollback();
+        }
+        catch (Throwable t) {
+            if (error == null) {
+                error = new RuntimeException("Exception during rollback");
+            }
+            throw error;
+        }
```

Comment:
I believe this is the same

```suggestion
            positionDeleteWriter.getWriter().rollback();
```

Reply:
The error message will be either "_Error rolling back write parquet to Hive_" or "_Error rolling back write to Hive_" for Parquet and Orc file writers respectively. Would that be okay?

Reply:
You can create a new error code in IcebergErrorCode, perhaps `ICEBERG_ROLLBACK_ERROR`.  Do not throw RuntimeException since it's unclassified, throw a new PrestoException with the new error code.